torch/_C/__init__.pyi.in

```diff
@@ -82,22 +85,46 @@ def _jit_get_operation(op_name: str) -> Callable: ...
-def _add_docstr(obj: T, doc_obj: str) -> T: ...
-def _from_dlpack(data: Any) -> Tensor: ...
-def _to_dlpack(data: Tensor) -> Any: ...
-def _set_backcompat_broadcast_warn(arg: _bool) -> None: ...
-def _get_backcompat_broadcast_warn() -> _bool: ...
-def _set_backcompat_keepdim_warn(arg: _bool) -> None: ...
-def _get_backcompat_keepdim_warn() -> _bool: ...
-def _is_xnnpack_enabled() -> _bool: ...
-def _get_mkldnn_enabled() -> _bool: ...
-def _set_mkldnn_enabled(arg: _bool) -> None: ...
-def _set_default_tensor_type(type) -> None: ...  # ick, what a bad legacy API
-def _set_default_dtype(d: _dtype) -> None: ...
-def _initExtension(shm_manager_path: str) -> None: ...
+def _initExtension(shm_manager_path: str) -> None: ...  # THPModule_initExtension
+def _autograd_init() -> _bool: ...  # THPAutograd_initExtension
+def _add_docstr(obj: T, doc_obj: str) -> T: ...  # THPModule_addDocStr
+def _init_names(arg: Sequence[Type]) -> None: ...  # THPModule_initNames
+def _has_distributed() -> _bool: ...  # THPModule_hasDistributed
+def _set_default_tensor_type(type) -> None: ...  # THPModule_setDefaultTensorType
+def _set_default_dtype(d: _dtype) -> None: ...  # THPModule_setDefaultDtype
+def _infer_size(arg1: Size, arg2: Size) -> Size: ...  # THPModule_inferSize
+def _crash_if_csrc_asan() -> _int: ...  # THPModule_crashIfCsrcASAN
+def _crash_if_csrc_ubsan() -> _int: ...  # THPModule_crashIfCsrcUBSAN
+def _crash_if_aten_asan() -> _int: ...  # THPModule_crashIfATenASAN
+def _show_config() -> str: ...  # THPModule_showConfig
+def _parallel_info() -> str: ...  # THPModule_parallelInfo
+def _set_backcompat_broadcast_warn(arg: _bool) -> None: ...  # THPModule_setBackcompatBroadcastWarn
+def _get_backcompat_broadcast_warn() -> _bool: ...  # THPModule_getBackcompatBroadcastWarn
+def _set_backcompat_keepdim_warn(arg: _bool) -> None: ...  # THPModule_setBackcompatKeepdimWarn
+def _get_backcompat_keepdim_warn() -> _bool: ...  # THPModule_getBackcompatKeepdimWarn
+def get_num_thread() -> _int: ...  # THPModule_getNumThreads
+def set_num_threads(nthreads: _int) -> None: ...  # THPModule_setNumThreads
+def get_num_interop_threads() -> _int: ...  # THPModule_getNumInteropThreads
+def set_num_interop_threads(nthreads: _int) -> None: ...  # THPModule_setNumInteropThreads
+def _get_cudnn_enabled() -> _bool: ...  # THPModule_userEnabledCuDNN
+def _set_cudnn_enabled(arg: _bool) -> None: ...  # THPModule_setUserEnabledCuDNN
+def _get_mkldnn_enabled() -> _bool: ...  # THPModule_userEnabledMkldnn
+def _set_mkldnn_enabled(arg: _bool) -> None: ...  # THPModule_setUserEnabledMkldnn
+def _get_cudnn_benchmark() -> _bool: ...  # THPModule_benchmarkCuDNN
+def _set_cudnn_benchmark(arg: _bool) -> None: ...  # THPModule_setBenchmarkCuDNN
+def _get_cudnn_deterministic() -> _bool: ...  # THPModule_deterministicCuDNN
+def _set_cudnn_deterministic(arg: _bool) -> None: ...  # THPModule_setDeterministicCuDNN
+# TODO: Is there a Capsule type in typing?  Doesn't seem so...
```

Comment:
Nope, doesn't exist. One discussion on python-dev where the suggestion is to add it to `types`, but wasn't done. From https://code.activestate.com/lists/python-dev/139675/:

>> - capsules
>> That one seems reasonable. But maybe there should just be a `types.Capsule` Type or `types.PyCapsule`, and then you can just check that the same as any other concrete type?

> But how often do you need to verify that something is a capsule, without knowing that it's the *right* capsule? At runtime, you'd usually use `PyCapsule_IsValid`, not `PyCapsule_CheckExacf`, right? So should the type checker be tracking the name too?